lib/hawkular/operations/operations_api.rb

```diff
@@ -33,9 +33,31 @@ class Client < Hawkular::BaseClient
     # helper for parsing the "OperationName=json_payload" messages
     class WebSocket::Frame::Data
       def to_msg_hash
-        chunks = split('=', 2)
+        operation_name, json = split('=', 2)
 
-        { operationName: chunks[0], data: JSON.parse(chunks[1]) }
+        # Check if there is a zip file following JSON.
+        # This check is done only in the first 100KB, hoping it's unlikely to
```

Comment:
Sounds reasonable, I don't think there is other way to know this. Unless there is a library that is able to parse only one json object from a string and ignore the rest.